internal/querycoordv2/handlers.go

```diff
@@ -0,0 +1,287 @@
+					Name: "",
+					ID:   int64(uniquegenerator.GetUniqueIntGeneratorIns().GetInt()),
+				},
+			})
+			continue
+		}
+
+		if metric.resp.Status.ErrorCode != commonpb.ErrorCode_Success {
+			log.Warn("invalid metrics of query node was found",
+				zap.Any("error_code", metric.resp.Status.ErrorCode),
+				zap.Any("error_reason", metric.resp.Status.Reason))
+			topo.ConnectedNodes = append(topo.ConnectedNodes, metricsinfo.QueryNodeInfos{
+				BaseComponentInfos: metricsinfo.BaseComponentInfos{
+					HasError:    true,
+					ErrorReason: metric.resp.Status.Reason,
+					Name:        metric.resp.ComponentName,
+					ID:          int64(uniquegenerator.GetUniqueIntGeneratorIns().GetInt()),
+				},
+			})
+			continue
+		}
+
+		infos := metricsinfo.QueryNodeInfos{}
+		err := metricsinfo.UnmarshalComponentInfos(metric.resp.Response, &infos)
+		if err != nil {
+			log.Warn("invalid metrics of query node was found",
+				zap.Error(err))
+			topo.ConnectedNodes = append(topo.ConnectedNodes, metricsinfo.QueryNodeInfos{
+				BaseComponentInfos: metricsinfo.BaseComponentInfos{
+					HasError:    true,
+					ErrorReason: err.Error(),
+					Name:        metric.resp.ComponentName,
+					ID:          int64(uniquegenerator.GetUniqueIntGeneratorIns().GetInt()),
+				},
+			})
+			continue
+		}
+		topo.ConnectedNodes = append(topo.ConnectedNodes, infos)
+	}
+}
+
+type metricResp struct {
+	resp *milvuspb.GetMetricsResponse
+	err  error
+}
+
+func (s *Server) tryGetNodesMetrics(ctx context.Context, req *milvuspb.GetMetricsRequest, nodes ...*session.NodeInfo) []*metricResp {
+	wg := sync.WaitGroup{}
+	ret := make([]*metricResp, 0, len(nodes))
+	retCh := make(chan *metricResp, len(nodes))
+	for _, node := range nodes {
+		node := node
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			resp, err := s.cluster.GetMetrics(ctx, node.ID(), req)
+			if err != nil {
+				log.Warn("failed to get metric from QueryNode",
+					zap.Int64("node-id", node.ID()))
+				return
+			}
+			retCh <- &metricResp{
+				resp: resp,
+				err:  err,
+			}
+		}()
+	}
+	wg.Wait()
+	close(retCh)
+	for resp := range retCh {
+		ret = append(ret, resp)
+	}
+	return ret
+}
+
+func (s *Server) fillReplicaInfo(replica *meta.Replica, withShardNodes bool) (*milvuspb.ReplicaInfo, error) {
+	info := utils.Replica2ReplicaInfo(replica.Replica)
+
+	channels := s.targetMgr.GetDmChannelsByCollection(replica.GetCollectionID())
+	if len(channels) == 0 {
+		msg := "failed to get channels, collection not loaded"
+		log.Warn(msg)
+		return nil, utils.WrapError(msg, meta.ErrCollectionNotFound)
+	}
+	var segments []*meta.Segment
+	if withShardNodes {
+		segments = s.dist.SegmentDistManager.GetByCollection(replica.GetCollectionID())
+	}
+
+	for _, channel := range channels {
+		leader, ok := s.dist.ChannelDistManager.GetShardLeader(replica, channel.GetChannelName())
+		var leaderInfo *session.NodeInfo
+		if ok {
+			leaderInfo = s.nodeMgr.Get(leader)
+		}
+		if leaderInfo == nil {
```

Comment:
leaderInfo is int64 type, should use !ok instead of  leaderInfo == nil

Reply:
`leaderInfo` is `*NodeInfo`, the dist may be updated later, so it's possible to get a `leaderID` from dist, but the node is offline (which leads to failed to get `leaderInfo` from node manager)